Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -22,239 +24,246 @@ class Client(BaseClient):
     implements get_token and get_events functions
     """
 
-    def __init__(self, verify, proxy, auth_url, gateway_url, base_url, client_id, client_secret, export_profile, headers=None):
-        self.auth_url = auth_url
-        self.gateway_url = gateway_url
-        self.client_id = client_id
-        self.client_secret = client_secret
+    def __init__(
+        self,
+        verify: bool,
+        proxy: bool,
+        auth_url: str,
+        base_url: str,
+        client_id: str,
+        client_secret: str,
+        export_profile: str,
+    ) -> None:
+        self.__auth_url = auth_url
+        self.__client_id = client_id
+        self.__client_secret = client_secret
         self.export_profile = export_profile
 
-        super().__init__(base_url=base_url, headers=headers, verify=verify, proxy=proxy)
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        self._headers = {'Authorization': f'Bearer {self._get_access_token()}'}
+
+    def _get_access_token(self) -> str:
+        """
+        Tries to find a valid token in integration context. If not found or expired, generates a new token
+        and sets it in the integration context.
 
-    def get_token(self):
+        Returns:
+            str: Access token.
+        """
         integration_context = get_integration_context()
         token = integration_context.get('token')
         valid_until = integration_context.get('valid_until')
         time_now = int(time.time())
-        if token and valid_until:
-            if time_now < valid_until:
-                # Token is still valid - did not expire yet
-                demisto.debug('Using cached token which is still valid')
-                demisto.debug(f'time-now: {time_now}\n valid token until: {valid_until}')
-                return token
-        sec = self.client_secret
+
+        if token and valid_until and time_now < valid_until:
+            # Token is still valid - did not expire yet
+            demisto.debug('Using cached token which is still valid')
+            demisto.debug(f'time-now: {time_now}\n valid token until: {valid_until}')
```

Comment:
```suggestion
            demisto.debug(f'Using cached token, still valid until: {valid_until}')
```

Reply:
Fixed in commit 582763c169821e56b0a063bc42e00239eb151efc